chainntnfs/btcdnotify/btcd.go

```diff
@@ -86,6 +86,9 @@ type BtcdNotifier struct {
 	// which the transaction could have confirmed within the chain.
 	confirmHintCache chainntnfs.ConfirmHintCache
 
+	// mempool gives the client access to transactions in the mempool.
+	mempool *chainntnfs.Mempool
```

Comment:
Ah I see this commit now. See the comment above having it be distinct from `RegisterSpend` ntfn. 

For the fast-settle path re Loop and other swap services I mentioned, you'd want the incoming link to register for this if the outgoing link hits the chain. The existing code path isn't sufficient, as the contract resolver will only check the witness beacon/cache once it needs to force close.

Reply:
So the current path is, the outgoing link hits the chain, extracts the preimage from blocks or mempool, adds it to the preimage beacon, and feeds the incoming link.